pkg/server/server.go

```diff
@@ -38,8 +43,11 @@ const (
 	defaultWebhookHTTPSPort = 9443
 	webhookPortEnvKey       = "CATTLE_PORT"
 	webhookURLEnvKey        = "CATTLE_WEBHOOK_URL"
+	allowedCNsEnv           = "ALLOWED_CNS"
 )
 
+var caFile = filepath.Join(os.TempDir(), "k8s-webhook-server", "client-certs", "ca.crt")
```

Comment:
Let's change this to store the cert outside of `/tmp`. Maybe `/var/lib/rancher/webhook/client-certs`?

@macedogm thoughts?

Reply:
Fully agree. `/tmp` is not ideal to store sensitive data, even if temporarily.

Reply:
@pjbgf / @macedogm 
I will note that this is the default dir we are already using to store certs used for our CAPI webhook
https://github.com/kubernetes-sigs/controller-runtime/blob/7cfd07f52ecec21599a686ab7ae305b98fff35de/pkg/webhook/server.go#L82C2-L82C2

https://github.com/rancher/webhook/blob/release/v0.4/pkg/capi/capi.go#L53

If we change this, then we will have users storing their certs in two separate locations.

Reply:
Thanks for the explanation @KevinJoiner and @cmurphy.

@cmurphy it's the signing CA used to sign which certificates?

Reply:
But just to confirm, wouldn't https://github.com/rancher/webhook/pull/273/files#diff-547361dafecac27b6e2da396d00d4c8c673b1fabda2bb3ad3a5860c6706aed2cR90 be a sensitive data?

Reply:
Maybe client-certs is a bad directory name, I was trying to distinguish it from the serving-certs. The only item in that directory should be the CA which signed the kube-apiserver client certs.

Reply:
correct

Reply:
@cmurphy thanks for the explanation. Yes, we got confused by the dir name, because it implied something sensitive, which is not the case then.

Reply:
Will change it, thanks for the feedback!

Reply:
Thanks for all the context @cmurphy @KevinJoiner, I wasn't aware of the capi/upstream. Let's keep it as is for the time being.